src/tests/homework1.spec.js

```diff
@@ -1,170 +1,142 @@
 
-    const nameInput = page.getByLabel('Jméno a příjmení');
-    const emailInput = page.getByLabel('Email');
-    const passwordInput = page.getByLabel('heslo');
-    const controlPasswordInput = page.getByLabel('Kontrola hesla');
-    const registerButton = page.getByRole('button', { name: 'Zaregistrovat' });
-    const logoutButton = page.locator('#logout-link');
+        //first registration + logout -> get certain email
+        const nameInput = page.getByLabel('Jméno a příjmení');
+        const emailInput = page.getByLabel('Email');
+        const registerButton = page.getByRole('button', { name: 'Zaregistrovat' });
+        const logoutButton = page.locator('#logout-link');
 
-    var randomName = faker.person.fullName();
-    await nameInput.fill(randomName);
+        var randomName = faker.person.fullName();
+        await nameInput.fill(randomName);
 
-    const randomEmail = faker.internet.email();
-    await emailInput.fill(randomEmail);
+        const randomEmail = faker.internet.email();
+        await emailInput.fill(randomEmail);
 
-    await passwordInput.fill('noveHeslo1');
-    await controlPasswordInput.fill('noveHeslo1');
+        await fillPassword(page);
 
-    await registerButton.click();
+        await registerButton.click();
 
-    await page.getByRole('button', { name: randomName }).click();
-    await logoutButton.click();
+        await page.getByRole('button', { name: randomName }).click();
+        await logoutButton.click();
 
-    //another registration with currently used email
-    await page.goto('/registrace');
+        //another registration with currently used email
+        await page.goto('/registrace');
 
-    randomName = faker.person.fullName();
-    await nameInput.fill(randomName);  //vyplním nově vygenerované jméno
-    await emailInput.fill(randomEmail);  //použiju stejný email jako při registraci výše
-    await passwordInput.fill('noveHeslo1');
-    await controlPasswordInput.fill('noveHeslo1');
-    await registerButton.click();
+        randomName = faker.person.fullName();
+        await nameInput.fill(randomName);  //vyplní nově vygenerované jméno
+        await emailInput.fill(randomEmail);  //použije stejný email jako při registraci výše
+        await fillPassword(page);
+        await registerButton.click();
 
 
-    await expect(page).toHaveURL(/.*registrace/); //ověřím, že jsem stále na stránce registrace
-    const toastMessage = page.locator('.toast-message');
-    const fieldError = page.locator('.invalid-feedback').locator('strong');
-    //ověřím chybové hlášky
-    await expect(toastMessage).toHaveText("Některé pole obsahuje špatně zadanou hodnotu");
-    await expect(fieldError).toHaveText("Účet s tímto emailem již existuje");
-    //await expect(page.getByText('Účet s tímto emailem již')).toBeVisible();
+        await expect(page).toHaveURL(/.*registrace/); //ověří, že je stále na stránce registrace
```

Comment:
Assertovat URL je fajn a v pohodě, ale nebylo by úplně od věci assertovat elementy. Že je zobrazený registrační formulář je důležitější.